### PR TITLE
map cantabular dataset dimensions and population types to the produce…

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

Trello cards:
 - https://trello.com/c/bSSV4HgX/1183-update-search-data-extractor-to-extract-dimensions-for-datasets
 - https://trello.com/c/BoGQFz9W/1184-update-search-data-extractor-to-extract-population-type-for-datasets

Main tasks:

- Extend producer kafka model with array of dimensions and 
- Implement the mapping of new fields, only for cantabular dataset types (`cantabular_flexible_table` and `cantabular_multivariate_table`)
- Ignore dimensions that where area_type == true
- Used hardcoded mapping of names and labels for population types

Extra bits:

- Refactored a bit the handler code to separate the legacy (zebedee) logic from the dataset logic.
- Upgrade to go v1.19
- Add Dockerfile.local and reflex so that the service can be deployed locally in `dp-compose` stacks

### How to review

Check if code changes make sense and unit tests pass and cover all possible new scenarios

### Who can review

Anyone.

### CODE REVIEW

File: handler/datasets.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	versionMetadata := models.CMDData{
+		UID:            generatedID,
+		URI:            parsedURI.Path,
+		VersionDetails: versionDetails,
+		DatasetDetails: datasetDetailsData,
+	}
+
+	datasetVersionMetadata := models.MapVersionMetadataToSearchDataImport(versionMetadata)
```
Comment (nshumoogum) [inline handler/datasets.go:118]: Suggestion: we could just pass the datasetMetadataPublished and cpEvent objects into this method and extend to handle all transformations

File: handler/datasets.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	dd.Type = t
+	log.Info(ctx, "identified dataset with cantabular type", log.Data{
+		"type":           t,
+		"num_dimensions": len(metadata.Dimensions)},
+	)
+
+	dd.Dimensions = []models.Dimension{}
+	for i := range metadata.Dimensions {
+		// Using pointers to prevent copying lots of data.
+		// TODO consider changing type to []*VersionDimension in dp-api-clients-go
```
Comment (nshumoogum) [inline handler/datasets.go:162]: We probably need to wait for new client planned to be created inside the dataset API - nothing to do here and happy for you to leave this comment in here for now

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: handler/datasets.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+	// Make a call to DatasetAPI
+	datasetMetadataPublished, err := h.DatasetCli.GetVersionMetadata(ctx, "", cfg.ServiceAuthToken, cpEvent.CollectionID, datasetID, edition, version)
+	if err != nil {
+		log.Error(ctx, "cannot get dataset published contents version %s from api", err)
+		return err
+	}
+
+	logData["content_published"] = datasetMetadataPublished
+	log.Info(ctx, "datasetAPI response ", logData)
```
[the remark this answers is not in the document]
Comment (DavidSubiros) [reply handler/datasets.go:69]: I can repalce it with a smaller log

File: handler/datasets.go (hunk 1 above)
Comment (DavidSubiros) [reply handler/datasets.go:118]: I'm restructuring the mapper a bit as well, there is no need to use the intermediate structs, and the mapping will be much more clearer. Doing the following:
1- Create the event in the handler, initialised with values that come directly from the cpEvent (the datasetID, edition, version, etc. )
2- call dataset api to obtain the metadata
3- map the metadata to the corresponding values in search data import event, without any intermediary structs (done in the models package, as mapper)

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**